fact/factdb/utils.py

```diff
@@ -42,13 +38,8 @@ def get_ontime_by_source(runtype=None, config=None):
     if runtype is not None:
         query = query.where(RunType.fruntypename == runtype)
 
-    query = (
-        query
-        .group_by(Source.fsourcename)
-        .dicts()
-        .naive()  # put selected foreign key attributes directly into the objects
-    )
-    df = pd.DataFrame(list(query))
+    query, params = query.group_by(Source.fsourcename).sql()
+    df = pd.read_sql(query, engine or create_factdb_engine(), params=params)
```

Comment:
wow ... this `engine or create_factdb_engine()` strongly reminds me of this `do() or die()` pattern known in php.
smells strange to me ... but looks awesome ...